lib/views/pr-commits-view.js

```diff
@@ -0,0 +1,143 @@
+import React, {Fragment} from 'react';
+import PropTypes from 'prop-types';
+import {graphql, createPaginationContainer} from 'react-relay';
+import {RelayConnectionPropType} from '../prop-types';
+import PrCommitView from './pr-commit-view';
+
+import {autobind} from '../helpers';
+
+const PAGE_SIZE = 50;
+
+export class PrCommitsView extends React.Component {
+  static propTypes = {
+    relay: PropTypes.shape({
+      hasMore: PropTypes.func.isRequired,
+      loadMore: PropTypes.func.isRequired,
+      isLoading: PropTypes.func.isRequired,
+    }).isRequired,
+    pullRequest: PropTypes.shape({
+      commits: RelayConnectionPropType(
+        PropTypes.shape({
+          commit: PropTypes.shape({
+            committer: PropTypes.shape({
+              avatarUrl: PropTypes.string.isRequired,
+              name: PropTypes.string.isRequired,
+              date: PropTypes.string.isRequired,
+            }),
+            messageBody: PropTypes.string,
+            messageHeadline: PropTypes.string.isRequired,
+            abbreviatedOid: PropTypes.string.isRequired,
+            url: PropTypes.string.isRequired,
+          }),
+        }),
+      ),
+    }),
+  }
+
+  constructor(props) {
+    super(props);
+    autobind(this, 'loadMore');
+  }
+
+  loadMore() {
+    this.props.relay.loadMore(PAGE_SIZE, () => {
+      this.forceUpdate();
+    });
+    this.forceUpdate();
+  }
+  render() {
+    return (
+      <Fragment>
+        <div className="github-PrCommitsView-commitWrapper">
+          {this.renderCommits()}
+        </div>
+        {this.renderLoadMore()}
+      </Fragment>
+    );
+  }
+
+  renderLoadMore() {
+    if (!this.props.relay.hasMore()) {
+      return null;
+    }
+    return (
+      <button className="github-PrCommitsView-load-more-button btn" onClick={this.loadMore}>Load more</button>);
+  }
+
+  renderCommits() {
+    return this.props.pullRequest.commits.edges.map(edge => {
+      const commit = edge.node.commit;
+      return (
+        <PrCommitView
+          key={commit.abbreviatedOid}
+          committerAvatarUrl={commit.committer.avatarUrl}
+          date={commit.committer.date}
+          messageBody={commit.messageBody}
+          messageHeadline={commit.messageHeadline}
+          abbreviatedOid={commit.abbreviatedOid}
+          url={commit.url}
+          committerName={commit.committer.name}
+        />);
+    });
+  }
+}
+
+export default createPaginationContainer(PrCommitsView, {
+  pullRequest: graphql`
+    fragment prCommitsView_pullRequest on PullRequest
+    @argumentDefinitions(
+      commitCount: {type: "Int!", defaultValue: 100},
+      commitCursor: {type: "String"}
+    ) {
+      url
+      commits(
+        first: $commitCount, after: $commitCursor
+      ) @connection(key: "prCommitsView_commits") {
+        pageInfo { endCursor hasNextPage }
+        edges {
+          cursor
+          node {
+            commit {
```

Comment:
This is just a nice to have -- we could extract this as a fragment on PrCommitView. See [timeline/commits-view.js](https://github.com/atom/github/blob/master/lib/views/timeline-items/commits-view.js#L85) and [timeline/commit-view.js](https://github.com/atom/github/blob/master/lib/views/timeline-items/commit-view.js#L69-L86) as an example.

This keeps our query fragments co-located with the views that use the resulting data. I think this may also allow relay to do some extra checking for us to make sure that data from our queries match with what is actually being used.